eZ/Publish/Core/Search/Legacy/Content/Common/Gateway/SortClauseHandler/Field.php

```diff
@@ -191,8 +174,95 @@ public function applyJoin(SelectQuery $query, SortClause $sortClause, $number)
                         $this->dbHandler->quoteColumn('version', $table),
                         $this->dbHandler->quoteColumn('current_version', 'ezcontentobject')
                     ),
-                    $languageExpression
+                    $this->getFieldCondition($query, $languageSettings, $table)
                 )
             );
     }
+
+    /**
+     * Returns a field language join condition for the given $languageSettings.
+     *
+     * @param \eZ\Publish\Core\Persistence\Database\SelectQuery $query
+     * @param array $languageSettings
+     * @param string $fieldTableName
+     *
+     * @return string
+     */
+    protected function getFieldCondition(SelectQuery $query, array $languageSettings, $fieldTableName)
```

Comment:
this seems to filter on ezcontentobject, so a bit strange to apply this in Field SortClause.